### ISSUE TITLE
Errors compiling

### ISSUE THREAD

**Issue by zezba9000** [ciplogic/CodeRefractor#11]
Just pulled the code and made no code changes.  First run I get the error:
"Calling the compiler failed: Index was out of range. Must be non-negative and less than the size of the collection"

Also why are you not just spitting out ".h" and ".cpp" sources so I can compile with any native compiler?  iOS uses Clang, Windows Phone uses VC++ ect

**Comment by bmustiata** [comment]
Hi zezba, thank you for your report.

Currently CodeRefractor is in the middle of some refactoring and stabilisation, and also moving to a new website. So I apologize for the unstable state of it. We're also looking on how we can get some integration testing up.

Regarding the compiler error, in the current branch, we're doing exactly that, see: https://github.com/ciplogic/CodeRefractor/commit/4bb90233c9c4ae8c489c274944be055e8eb2265a, At the moment we're outputting a single .cpp file. This is an actual bug in the transpiler, and the error message should be updated.

**Comment by ciplogic** [comment]
@zezba9000 Originally the compielr was using a hardcoded DevC++ GCC distribution.

The transpiler just does this (look for output.cpp file), so you can include it into VC++ (you should have in the bin folder a VC++ project which includes Output.cpp, and also output.dev (devC++) if you want to integrate it with GCC. 

I was never able to work with Clang (at least on Windows) in a stable fashion, but if you have any issues with generated code, I will be glad to help you address those issues.

**Comment by zezba9000** [comment]
I have a couple questions if thats ok.

1) How portable will the converter be?  Can I target GCC 3.x and above or is this C11 only?
1.1) What are the targets or how easy will it be able to target OS types that are not GNU Linux, Win32 or OSX, such as Embedded platforms or others?  Does this rely on very portable standards like std or something more?

2) What is the GC thats used?  Sorry if I missed that somewhere.

**Comment by ciplogic** [comment]
@zezba9000 
For now is C++ 11 only because of using smart-pointers. I think the code should work into a GCC 4.5 or 4.6 as mode: "0x". This also removes the need of GC (yes, you can still have circular references).

CR compiler does a lot of analysis to make sure it reduces the usage of smart-pointers where is not needed so there is not a lot of reference counting updates.

"Does this rely on very portable standards like std or something more?"
Yes, excluding if you use PlatformInvoke (which is platform dependent), the generated code (even is a lot of goto's and such) is using only standards. If you find any breakage or any platform independent fixes, please notify us and we will be glad to address them.

"... such as Embedded platforms or others?"
bmustiata works especially into trying to address compatibility issues about embedded platforms.

In short, CR can swap the runtime and can be guided to resolve (very similar in a way with Java's InvokeDynamic) for a special runtime/platform even it doesn't support it. The single thing to solve on your end is to define a "platform stub" (a dll that gives public interface with no implementation) and a resolver that will write your platform specific code. If there are any bugs around it, please notify us.
